src/main/java/com/griddynamics/cd/nrp/internal/uploading/impl/ArtifactUpdateApiClientImpl.java

```diff
@@ -51,13 +48,23 @@
     private ConfigurationsManager configurationsManager;
 
     /**
+     * Queue for async requests
+     */
+    private BlockingQueue<Runnable> queue;
+
+    /**
      * ExecutorService shares between clients. All treads are created in the same executor
      */
-    private ExecutorService asyncRequestsExecutorService = Executors.newFixedThreadPool(10);
+    private ExecutorService asyncRequestsExecutorService;
 
     @Inject
     public ArtifactUpdateApiClientImpl(ConfigurationsManager configurationsManager) {
         this.configurationsManager = configurationsManager;
+        this.queue = new LinkedBlockingQueue<>(configurationsManager.getConfiguration().getRequestsQueueSize());
+        this.asyncRequestsExecutorService = new ThreadPoolExecutor(
+                configurationsManager.getConfiguration().getRequestsSendingThreadsCount(),
+                configurationsManager.getConfiguration().getRequestsSendingThreadsCount(),
+                30, TimeUnit.SECONDS, queue);
```

Comment:
- 30 secs seems to be too low, WDYT?
- The default Rejection Execution Handler is the one that simply throws exception. Is that what we _actually_ want when there is no place for a new request? If so, I'd document this explicitly - the decision itself and the reason why we chose it.